core/src/main/java/org/elasticsearch/search/suggest/SuggestionBuilder.java

```diff
@@ -197,7 +197,18 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
     protected abstract SuggestionBuilder<T> innerFromXContent(QueryParseContext parseContext, String name) throws IOException;
 
-    protected abstract SuggestionContext build(QueryShardContext context) throws IOException;
+    public SuggestionContext build(QueryShardContext context) throws IOException {
+        SuggestionContext suggestionContext = innerBuild(context);
+        // copy over common settings to each suggestion builder
+        populateCommonFields(context.getMapperService(), suggestionContext);
```

Comment:
That's certainly one option.  The only thing I don't like about that option is that every subclass must remember to call `populateCommonFields` in its own implementation of `innerBuild`.  It doesn't seem the best separation of duties between the super class and the subclass.  I lean toward the option of having a separate `newInstance` method that each subclass implements, so goes something like:
`
SuggestionContext suggestionContext = newInstance();
populateCommonFields(context.getMapperService(), suggestionContext);
innerBuild(suggestionContext);
return suggestionContext;
`

Thoughts?  I'm open to other suggestions if others think the other way is better.